contrib/submit-protocol/src/App/MarketplaceForm.tsx

```diff
@@ -0,0 +1,190 @@
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React from "react";
+import {
+  Callout, DefaultButton, Dropdown, DropdownMenuItemType,
+  Fabric, IDropdownOption, Stack, TextField,
+  mergeStyleSets,
+} from "office-ui-fabric-react";
+
+enum MarketplaceUriType {
+  None = "None",
+  GitHub = "GitHub",
+}
+
+interface IMarketplaceProps {
+  uri?: string;
+  uriType?: MarketplaceUriType | string;
+  onSelectProtocol: ((text: string) => void);
+  disabled?: boolean;
+}
+
+interface IMarketplaceState {
+  uri: string;
+  uriType: MarketplaceUriType;
+  selectedProtocol?: IDropdownOption;
+  protocolOptions: IDropdownOption[];
+  uriConfigCallout: boolean;
+}
+
+const styles = mergeStyleSets({
+  dropdown: {
+    width: 240,
+  },
+});
+
+export default class MarketplaceForm extends React.Component<IMarketplaceProps, IMarketplaceState> {
+  public state = {
+    uri: this.props.uri || "",
+    uriType: this.props.uriType as MarketplaceUriType || MarketplaceUriType.None,
+    protocolOptions: [
+      { key: "None", text: "None" },
+      { key: "jobsDivider", text: "-", itemType: DropdownMenuItemType.Divider },
+      { key: "jobsHeader", text: "Protocol Jobs", itemType: DropdownMenuItemType.Header },
+    ],
+    uriConfigCallout: false,
+  };
+
+  private uriConfigCalloutBtn = React.createRef<HTMLDivElement>();
+
+  public componentDidMount() {
+    this.setProtocolOptions(null as any);
+  }
+
+  public render() {
+    return (
+      <Fabric>
+        <Stack gap={5} horizontal={true} verticalAlign="center">
+          <Stack>
+            <Dropdown
+              className={styles.dropdown}
+              placeholder="Select a protocol config file"
+              options={this.state.protocolOptions}
+              onChange={this.selectProtocol}
+              disabled={this.props.disabled}
+            />
+          </Stack>
+          <Stack maxWidth="5px">
+            <div ref={this.uriConfigCalloutBtn}>
+              <DefaultButton
+                onClick={this.onConfigCallout}
+                iconProps={{iconName: "ConfigurationSolid"}}
+                text="URI"
+                disabled={this.props.disabled}
+              />
+            </div>
+            <Callout
+              role="alertdialog"
+              target={this.uriConfigCalloutBtn.current}
+              onDismiss={this.closeConfigCallout}
+              setInitialFocus={true}
+              hidden={!this.state.uriConfigCallout}
+            >
+              <Stack padding={20}>
+                <TextField
+                  label="Marketplace URI"
+                  prefix="https://api.github.com/repos/"
+                  description="Endpoint should be :owner/:repo/contents/:path?ref=:branch"
```

Comment:
It's better to allow user to set owner / repo / path / branch seperately

Reply:
Because there may be other sources, simplify it to a raw input and user should input the whole uri.
This is a temporal solution and another forms may be used to support different sources.